### ISSUE TITLE
[21] Code assist does not work inside String template

### ISSUE THREAD

**Issue by jarthana** [eclipse-jdt/eclipse.jdt.core#1685]
This is a follow up bug after pull #1513.

Expressions inside a String templates are not recognized by code selection or completion engines. This is due to the fact the embedded expressions are parsed using a sub parser (than the original one) and there is no state flow between these two parsers. As per @srikanth-sankaran, there's a prior art for such cases and will work on this.

**Comment by srikanth-sankaran** [comment]
> This is a follow up bug after pull #1513.
> 
> Expressions inside a String templates are not recognized by code selection or completion engines. This is due to the fact the embedded expressions are parsed using a sub parser (than the original one) and there is no state flow between these two parsers. As per @srikanth-sankaran, there's a prior art for such cases and will work on this.

Correction: There is prior lack of art :-)

**Comment by srikanth-sankaran** [comment]
I'll keep this ticket for code selection support. And once this is merged will work on a separate PR/ticket for code completion
